back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -3,17 +3,56 @@
 from peewee import JOIN, SQL, fn
 
 from ...db import db
-from ...enums import HumanGender, TaggableObjectType
+from ...enums import BoxState, HumanGender, TaggableObjectType
 from ...models.definitions.beneficiary import Beneficiary
 from ...models.definitions.box import Box
+from ...models.definitions.history import DbChangeHistory
 from ...models.definitions.location import Location
 from ...models.definitions.product import Product
 from ...models.definitions.product_category import ProductCategory
+from ...models.definitions.size import Size
 from ...models.definitions.tag import Tag
 from ...models.definitions.tags_relation import TagsRelation
+from ...models.definitions.transaction import Transaction
 from ...models.utils import convert_ids
 
 
+def _generate_dimensions(*names, facts):
+    """Return a dictionary holding information (ID, name) about dimensions with
+    specified names.
+    """
+    dimensions = {}
+
+    if "product" in names:
+        product_ids = {f["product_id"] for f in facts}
+        dimensions["product"] = (
+            Product.select(Product.id, Product.name)
```

Comment:
Here, is something we might want to talk about. A Product always has a gender. If we do not add this to the dimension, many values will have the same name.

For example:
Usually there is a product "Winter Jacket" for Male and Female. They are two separate products, but have the same name.

Proposal:
Either we concatenate the Gender name to the Product name or we create an extra dimension for gender.

If we add gender as a dimension, we can also think about grouping the facts not by product id, but by product name.

Reply:
Would it also work to return `gender` in the product dimension table? I think it gives most flexibility, and FE can then concatenate product and gender to their liking.

Reply:
addressed in #998